tools/ci-build/publisher/src/package.rs

```diff
@@ -255,38 +340,60 @@ fn read_package(path: &Path, manifest_bytes: &[u8]) -> Result<Option<Package>> {
 /// Validates that all of the publishable crates use consistent version numbers
 /// across all of their local dependencies.
 fn validate_packages(packages: &[Package]) -> Result<()> {
-    let mut versions: BTreeMap<String, Version> = BTreeMap::new();
-    let track_version = &mut |handle: &PackageHandle| -> Result<(), Error> {
-        if let Some(version) = versions.get(&handle.name) {
+    let mut versions: BTreeMap<String, LenientVersion> = BTreeMap::new();
+    let mut version_requirements: BTreeMap<String, LenientVersion> = BTreeMap::new();
+    let track_version = &mut |handle: &PackageHandle,
+                              map: &mut BTreeMap<String, LenientVersion>|
```

Comment:
Doesn't look like this refers to anything outside of the closure, so you can just declare it as a normal nested function.

Reply:
Addressed in 1597129.